dot/state/transaction.go

```diff
@@ -58,5 +66,47 @@ func (s *TransactionState) RemoveExtrinsicFromPool(ext types.Extrinsic) {
 
 // AddToPool adds a transaction to the pool
 func (s *TransactionState) AddToPool(vt *transaction.ValidTransaction) common.Hash {
+	s.notifyStatus(transaction.StatusNotification{Ext: vt.Extrinsic, Status: transaction.Future.String()})
 	return s.pool.Insert(vt)
 }
+
+// GetStatusNotifierChannel creates and returns a status notifier channel.
+func (s *TransactionState) GetStatusNotifierChannel() chan transaction.StatusNotification {
```

Comment:
it might be nice to also have this accept an `[]types.Extrinsic` of extrinsics to watch, so it only gets notified when those update, as opposed to all transactions pushing a notifications

Reply:
addressed in the latest commit